magento/pwa-studio
# Limit CSS Modules to *.module.css files

## Description

It's difficult to add new loaders and import custom CSS files when our webpack configuration blindly applies `modules: true` to all CSS files. The existing exception for `node_modules` is insufficient; a filename pattern would help. The `*.module.css` pattern is standard (see [NextJS docs](https://nextjs.org/docs/basic-features/built-in-css-support#adding-component-level-css)).

- Update webpack config to only set `modules: true` for `*.module.css` files (plus opt-ins)
- Rename all CSS files that expect `modules: true`
- Update references to old filenames, including imports

## Related Issue

PWA-2117

## Acceptance

### Verification Stakeholders

- @jcalcaben
- @tjwiebell 

### Specification

## Verification Steps

- Ensure the app builds with no errors or warnings
- Ensure Tailwind generated classnames are not transformed by CSS Modules
- Ensure there are no visual regressions (nothing changed)

#### Test scenario(s) for direct fix/feature

#### Test scenario(s) for any existing impacted features/areas

#### Test scenario(s) for any Magento Backend Supported Configurations

#### Is Browser/Device testing needed?

#### Any ad-hoc/edge case scenarios that need to be considered?

## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)

Hundreds of CSS files have been renamed and their components updated to point to the new filenames, so everything here requires a major version.

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
